pkg/ebpf/c/common/buffer.h

```diff
@@ -98,6 +104,45 @@ statfunc int save_bytes_to_buf(args_buffer_t *buf, void *ptr, u32 size, u8 index
     return 0;
 }
 
+statfunc int load_str_from_buf(args_buffer_t *buf, char *str, u8 index)
+{
+    u16 offset;
+    u32 size;
+
+    // skip if index is not in buffer
+    if (index > buf->argnum)
+        return 0;
+
+    offset = buf->args_offset[index];
+
+    if (offset == INVALID_ARG_OFFSET)
+        return 0;
+
+    // Ensure there is enough space for read index (u8)
+    if (offset + sizeof(u8) > ARGS_BUF_SIZE)
+        return 0;
+
+    // Skip index
+    offset += sizeof(u8);
+
+    // Copy the size
+    __builtin_memcpy(&size, &(buf->args[offset]), sizeof(u32));
+
+    // Check size against max length, leaving space for null terminator
+    if (size > MAX_DATA_FILTER_STR_SIZE - 1)
```

Comment:
If greater should we abort filtering or consider the string filter up to the max?

If aborting is the answer, could we avoid this check by enforcing the set of the filter in userspace? If larger than max, I suppose it shouldn't be set at first.

Reply:
@geyslan the `load_str_from_buf` was complete reworded ... and its aligned with your doubt... if you don't mind, pls take a look on the function again